nkms/blockchain/eth/nkms_eth/blockchain.py

```diff
@@ -0,0 +1,83 @@
+from abc import ABC
+
+from nkms_eth.config import EthereumConfig
+
+
+class TheBlockchain(ABC):
+    """
+    http://populus.readthedocs.io/en/latest/config.html#chains
+
+    mainnet: Connects to the public ethereum mainnet via geth.
+    ropsten: Connects to the public ethereum ropsten testnet via geth.
+    tester: Uses an ephemeral in-memory chain backed by pyethereum.
+    testrpc: Uses an ephemeral in-memory chain backed by pyethereum.
+    temp: Local private chain whos data directory is removed when the chain is shutdown. Runs via geth.
+    """
+
+    _network = NotImplemented
+    _default_timeout = NotImplemented
+    __instance = None
+
+    test_chains = ('tester', )
+    transient_chains = test_chains + ('testrpc', 'temp')
+    public_chains = ('mainnet', 'ropsten')
+
+    class IsAlreadyRunning(RuntimeError):
+        pass
+
+    def __init__(self, eth_config: EthereumConfig):
+        """
+        Configures a populus project and connects to blockchain.network.
+        Transaction timeouts specified measured in seconds.
+
+        http://populus.readthedocs.io/en/latest/chain.wait.html
+
+        """
+
+        # Singleton
+        if TheBlockchain.__instance is not None:
+            message = '{} is already running on {}. Use .get() to retrieve'.format(self.__class__.__name__, self._network)
+            raise TheBlockchain.IsAlreadyRunning(message)
+        TheBlockchain.__instance = self
+
+        self._eth_config = eth_config
+        self._chain = eth_config.provider  # TODO
+
+    @classmethod
+    def get(cls):
+        if cls.__instance is None:
+            class_name = cls.__name__
+            raise Exception('{} has not been created.'.format(class_name))
+        return cls.__instance
+
+    def disconnect(self):
+        self._chain.__exit__(None, None, None)
```

Comment:
Perhaps, makes sense to specify keyword arguments. Not super-important here, but I'm much more comfortable when I know what these Nones actually stand for